api/server/router.go

```diff
@@ -61,6 +89,19 @@ func (r *router) GetHandler(base, endpoint string) (http.Handler, error) {
 	return handler, nil
 }
 
+func (r *router) AddHeaderRoute(route string, handler http.Handler) bool {
+	r.routeLock.Lock()
+	defer r.routeLock.Unlock()
```

Comment:
When we are reading from `headerRoutes`, we hold `r.lock`. When we are writing to `headerRoutes` we are holding `r.routeLock`. I think we need to hold `r.lock` here right?
```suggestion
	r.lock.Lock()
	defer r.lock.Unlock()
```